plugin/announce.lua

```diff
@@ -145,26 +149,23 @@ minetest.register_chatcommand("server-announce", {
 	func = function(name, param)
 		if param and param ~= "" then
 			if minetest.check_player_privs(name, { ban = true }) then
-				manage_announcements(name, param)
-			else
-				minetest.chat_send_player(name, "Required privileges for arguments: ban")
+				return manage_announcements(param)
 			end
+			return false, "Required privileges for arguments: ban. You can run this command without arguments."
 		elseif announce_message then
-			beerchat.send_message(name, announce_message, "server")
-		else
-			minetest.chat_send_player(name, "No active announcements.")
+			return true, announce_message
 		end
+		return true, "No active announcements."
 	end
 })
 
 minetest.register_on_joinplayer(function(player)
 	if announce_message then
-		beerchat.send_message(player:get_player_name(), announce_message, "server")
+		minetest.chat_send_player(player:get_player_name(), announce_message)
```

Comment:
This affects how messages are delivered. Previous `beerchat.send_message` will call through `before_send` event callbacks and will not deliver message if any of those event handlers would deny message delivery.

I think it is still better to not go through registered callbacks with system messages, if there's need for per player configuration (#49 for example) then I think it is better to directly implement it here.

Anyone know if changing this might cause any issues with some special feature misuse?